### ISSUE TITLE
Identity provider sync mode: incorrect selection in case of null

### ISSUE THREAD

**Issue by lrozenblyum** [keycloak/keycloak#26019]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Identity provider sync mode shows 'Import' selection when the value is not stored on the backend.
![изображение](https://github.com/keycloak/keycloak/assets/1285898/2509d0f0-9e4e-4798-ab5e-e8ca021a6614)

However backend uses the `Legacy` value in this case (see `org.keycloak.models.IdentityProviderModel.getSyncMode()`)

### Version

22.0.5, 23.0.3

### Expected behavior

UI should show the select value for 'Sync mode' synchronous with backend (Legacy)

### Actual behavior

UI shows 'Import' as the selected value for 'Sync mode' which is inconsistent with the actual backend value.

### How to Reproduce?

2 ways to reproduce:

1) Create an identity provider via UI, save it ONCE. 
NOTE: During saving there is no possibility to select the sync mode (why?), so it's not saved to the database.
Then check the UI, it shows 'Import' as the selected value, but the backend acts in 'Legacy' mode

2) Update from Keycloak 8.0.1 to the actual version. No 'syncMode' properties exist for current identity providers.
Reason: https://github.com/keycloak/keycloak/pull/6705 didn't introduce any Liquibase migrations.

### Anything else?

If we create an identity provider from UI and save it twice, then the UI value is synchronous with backend (because UI already renders the 'Sync mode' selectbox and second save stores it to the db.

**Comment by lrozenblyum** [comment]
@ssilvert yes I agree, it will solve the issue for the newly created identity providers (**the 1'st way of reproduce**). Also it will improve UX I think - it's somehow unlogical that we need to save most fields at first and then some extra fields are shown just after save on the same form.

About the migrated data (**the 2'nd way to reproduce**) I see several possibilities how to fix
1) a Liquibase migration that fills in the correct default values for sync mode in case they're null
2) let backend return the same default value to UI that it uses in case when null is stored in db
3) let the UI select the same default value in the select-box as backend does when null is provided by backend 
any solution from 1)-3) would fix the 2'nd way of reproducing the issue I think.

General thoughts unrelated to this specific issue: in the long-term perspective, for new fields/features, maybe it would be a good idea to always provide the new default values in the Liquibase migrations for existing data to avoid this type of issues in future.

**Comment by ssilvert** [comment]
OK.  I think we can fix the first one pretty easily in the UI.

> 
> About the migrated data (**the 2'nd way to reproduce**) I see several possibilities how to fix
> 
> 1. a Liquibase migration that fills in the correct default values for sync mode in case they're null
> 2. let backend return the same default value to UI that it uses in case when null is stored in db
> 3. let the UI select the same default value in the select-box as backend does when null is provided by backend
>    any solution from 1)-3) would fix the 2'nd way of reproducing the issue I think.
> 

For this part, we will need to run this by the core team to implement your first or second suggestion.  I don't want the UI to make assumptions about what the back end is doing.  So I'd rather not do #3.

> General thoughts unrelated to this specific issue: in the long-term perspective, for new fields/features, maybe it would be a good idea to always provide the new default values in the Liquibase migrations for existing data to avoid this type of issues in future.

I agree.  It's always better to have a default whenever possible instead of `null`.
